# Exploratory analysis
This notebook briefly explores and evaluates the datasets that will be used in this project. Based on the findings, 
we will preprocess the datasets and thereafter conduct the analysis.

### Notebook environment setup

In [1]:
# Import packages and libraries
import pandas as pd

### Load and evaluate data

In [2]:
# Load dataset
df = pd.read_csv("../data/raw/big_cities_demographic_indicators-2026-06-15.csv")
df.info()
df.head()

<class 'pandas.DataFrame'>
RangeIndex: 3657 entries, 0 to 3656
Data columns (total 13 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   Demographic_Indicator                       3657 non-null   str    
 1   Year                                        3657 non-null   int64  
 2   Gender                                      3657 non-null   str    
 3   Race_Ethnicity                              3657 non-null   str    
 4   Value                                       3657 non-null   float64
 5   Lower_Limit_95_Percent_Confidence_Interval  2650 non-null   float64
 6   Upper_Limit_95_Percent_Confidence_Interval  2650 non-null   float64
 7   Place                                       3657 non-null   str    
 8   State                                       3510 non-null   str    
 9   State_Abbreviation                          3510 non-null   str    
 10  BCHC_Requested_Methodol

,Demographic_Indicator,Year,Gender,Race_Ethnicity,Value,Lower_Limit_95_Percent_Confidence_Interval,Upper_Limit_95_Percent_Confidence_Interval,Place,State,State_Abbreviation,BCHC_Requested_Methodology_For_Data,Data_Sources,Methods_And_Data_Notes
0,"Infant Mortality Rate (Per 1,000 live Birth)",2012,Both,Hispanic,5.5,NaN,NaN,Chicago,Illinois,IL,"Infant Deaths per 1,000 live Birth.","CDC, National Center for Health Statistics, Di...",NaN
1,"Infant Mortality Rate (Per 1,000 live Birth)",2012,Both,Black,12.7,NaN,NaN,Chicago,Illinois,IL,"Infant Deaths per 1,000 live Birth.","CDC, National Center for Health Statistics, Di...",NaN
2,"Infant Mortality Rate (Per 1,000 live Birth)",2012,Both,All,7.4,NaN,NaN,Chicago,Illinois,IL,"Infant Deaths per 1,000 live Birth.","CDC, National Center for Health Statistics, Di...",NaN
3,"Infant Mortality Rate (Per 1,000 live Birth)",2012,Both,Asian/PI,3.3,NaN,NaN,Chicago,Illinois,IL,"Infant Deaths per 1,000 live Birth.","CDC, National Center for Health Statistics, Di...",NaN
4,"Infant Mortality Rate (Per 1,000 live Birth)",2011,Female,All,3.5,2.0,5.7,Portland (Multnomah County),Oregon,OR,"Infant Deaths per 1,000 live Birth.","Oregon Public Health Assessment Tool, Linked B...",NaN


Upon loading the dataset, we find the following characteristics:
- Data types: float64, int64, str
- Shape: 16 columns, 3657 rows
- Memory: 371.5KB

In [4]:
df.isna().sum()

Demographic_Indicator                            0
Year                                             0
Gender                                           0
Race_Ethnicity                                   0
Value                                            0
Lower_Limit_95_Percent_Confidence_Interval    1007
Upper_Limit_95_Percent_Confidence_Interval    1007
Place                                            0
State                                          147
State_Abbreviation                             147
BCHC_Requested_Methodology_For_Data              0
Data_Sources                                   117
Methods_And_Data_Notes                         268
dtype: int64

Further investigating, we see a disparity in completeness between different columns. Demographic indicator, year,
gender, race & ethinciity, value, place, and BCHC methodology requests all showcase 100% completeness. On the contrary,
confidence
interval (lower and upper), state & state abreviation, data sources, and methods & data notes are missing in some rows.

We are primarily interested in further exploring why some states may be missing from their respective entries, as
well as data sources. Furthermore, checking to see how the lack of confidence intervals may impact data reliability
is also important.